### PR TITLE
[14.0][MIG] - membership_extension

### PR DESCRIPTION
**Pull request by sbejaoui**

### CODE REVIEW

File: membership_extension/models/membership_line.py [hunk 1]
```diff
@@ -42,13 +42,17 @@ def _onchange_membership_date(self):
...
                 line.state = line.state or "none"
-            if line.account_invoice_id.invoice_payment_state == "paid":
-                line.state = "paid"
+            elif (
+                line.account_invoice_id.state == "posted"
+                and line.account_invoice_id.payment_state == "reversed"
+            ):
+                line.state = "canceled"
+            else:
+                super(MembershipLine, line)._compute_state()
```
Comment (chienandalu) [inline membership_extension/models/membership_line.py:55]: Looking in the core method, I wonder if this override is needed anymore: https://github.com/odoo/odoo/blob/14.0/addons/membership/models/membership.py#L48-L78

File: membership_extension/tests/test_membership.py [hunk 2]
```diff
@@ -363,19 +359,17 @@ def test_invoice(self):
...
         reversal = move_reversal.reverse_moves()
         refund = self.env["account.move"].browse(reversal["res_id"])
-        refund.journal_id.update_posted = True
-        refund.action_post()  # validate refund
         self.assertEqual("canceled", line.state)
         refund.button_cancel()
         self.assertEqual("paid", line.state)
         refund.button_draft()
-        self.assertEqual("paid", line.state)
+        self.assertEqual("invoiced", line.state)
```
Comment (chienandalu) [inline membership_extension/tests/test_membership.py:372]: Was it wrong before?

File: membership_extension/models/membership_line.py (hunk 1 above)
Comment (sbejaoui) [reply membership_extension/models/membership_line.py:55]: The case of invoice refunded is not covered in core method and it fixes the possible issue when `account_invoice_id`  is null

https://github.com/odoo/odoo/blob/ff6d0b16cd32a5ed56a2cc039f5226dc56f6db6a/addons/membership/models/membership.py#L48

File: membership_extension/tests/test_membership.py (hunk 2 above)
Comment (sbejaoui) [reply membership_extension/tests/test_membership.py:372]: I am not sure how it was before, but now when the refund is canceled the original invoice is not automatically reconciled with the original payment.

**Comment by sbejaoui** [comment]
Travis is red due to failing test in 14.0, I created an issue for it #109

File: membership_extension/tests/test_membership.py (hunk 2 above)
Comment (rousseldenis) [reply membership_extension/tests/test_membership.py:372]: I think this is due to payment refactor.

**Comment by rousseldenis** [comment]
@pedrobaeza @sbejaoui This one is green now.

**Review by rousseldenis — APPROVED** [review]
Code review

**Review by victoralmau — APPROVED** [review]
Code review OK.

**Pull request merged**